softwareComponents/simplesimClientLib/src/changecolor.cpp

```diff
@@ -0,0 +1,104 @@
+#include "changecolor.hpp"
+
+#include <iostream>
+
+#include "ui_changecolor.h"
+
+using rofi::simplesim::ChangeColor;
+
+ChangeColor::ChangeColor( QWidget * parent, size_t size )
+        : _ui( std::make_unique< Ui::ChangeColor >() ), _parent( parent ), _toColor( size, false )
+{
+    _ui->setupUi( this );
+    _ui->plainTextEdit->hide();
+
+    connect( _ui->buttonBox, SIGNAL( rejected() ), this, SLOT( hide() ) );
+    connect( _ui->buttonBox, SIGNAL( accepted() ), this, SLOT( accept() ) );
+    connect( _ui->toolButton, SIGNAL( clicked() ), this, SLOT( toggleHelp() ) );
+}
+
+ChangeColor::~ChangeColor() {}
+
+struct syntaxError : std::exception {};
+struct rangeError : std::exception {};
```

Comment:
There is std::out_of_range - which can be already thrown by your code because of `std::stoi`.